ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -404,20 +410,24 @@ func (scbuilder *SiteConfigBuilder) getExtraManifest(dataMap map[string]interfac
 
 	// Adding workload partitions MC only for SNO clusters.
 	if clusterSpec.ClusterType == SNO && len(clusterSpec.Nodes) > 0 {
-		cpuSet := clusterSpec.Nodes[0].Cpuset
-		if cpuSet != "" {
-			k, v, err := scbuilder.getWorkloadManifest(cpuSet)
-			if err != nil {
-				errStr := fmt.Sprintf("Error could not read WorkloadManifest %s %s\n", clusterSpec.ClusterName, err)
-				return dataMap, errors.New(errStr)
-			} else {
-				data, err := addZTPAnnotationToManifest(v.(string))
+		for node := range clusterSpec.Nodes {
```

Comment:
Because we're now looping over all the nodes, the check for `&& len(clusterSpec.Nodes) > 0` is now redundant and can be removed.